### PR TITLE
DEPR: Timedelta.freq, Timedelta.is_populated

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pyx [hunk 1]
```diff
@@ -826,13 +826,31 @@ cdef _to_py_int_float(v):
...
+    #    bint _is_populated  # are my components populated
     #    int64_t _d, _h, _m, _s, _ms, _us, _ns
 
     # higher than np.ndarray and np.matrix
     __array_priority__ = 100
 
+    @property
+    def freq(self):
+        # GH#46430
+        warnings.warn(
```
Comment (jreback) [inline pandas/_libs/tslibs/timedeltas.pyx:838]: what does this do if its called? return None?

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
is there an issue for this (don't create one) but IIRC there is?

**Comment by jbrockmendel** [comment]
> is there an issue for this (don't create one) but IIRC there is?

No, @mroeschke suggested it in #46420

File: pandas/_libs/tslibs/timedeltas.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timedeltas.pyx:838]: correct

Comment (jreback) [reply pandas/_libs/tslibs/timedeltas.pyx:838]: can you make that explcit (and add the annotation)

**Review by mroeschke — APPROVED** [review]
LGTM pending green

File: pandas/_libs/tslibs/timedeltas.pyi [hunk 2]
```diff
@@ -146,3 +146,7 @@ class Timedelta(timedelta):
     def __hash__(self) -> int: ...
     def isoformat(self) -> str: ...
     def to_numpy(self) -> np.timedelta64: ...
+    @property
+    def freq(self) -> None: ...
+    @property
+    def is_populated(self) -> bool: ...
```
Comment (twoertwein) [inline pandas/_libs/tslibs/timedeltas.pyi:152]: Could also remove the type annotation to not encourage deprecated behavior.

For all the overloads that depend on keyword-only arguments, the type annotations already suggest that positional arguments are no longer supported (even though the implementation is still in the deprecation phase).

Comment (jreback) [reply pandas/_libs/tslibs/timedeltas.pyi:152]: yeah we don't really have a policy here for deprecated functions. i could go either way.

**Review by jreback — APPROVED** [review]

**Pull request merged**